parallaxinc/Parallax-IDE
# Identify & Replace Keypress

#### What's this PR do?

Adds keypress functionality for identify and replace. Also refactors dependent functionality into Flux.
#### Where should the reviewer start?

Pull this branch.
npm run build
#### How should this be manually tested?

Load a file with some text. Press CTRL-F4 to test the CodeMirror replace functionality. Press enter to replace as matches are found.
NOTE: for OSX users - CTRL-F4 is used by the OS to move window focus. To test this, you will need to disable the OSX keyboard shortcut. System Preferences > Keyboard > Shortcuts > Keyboard.

Press CTRL-I or F6 to trigger the download modal to pop up. Try downloading code to the STAMP. Try to hit Cancel and the ESC key in the download modal.

Test deleting files and creating files and saving. Ensure delete functionality is working as it was also moved to Flux.
#### Any background context you want to provide?

NOTE: Still need to test the download functionality with a STAMP. It along with everything else should be working.

To enable the shortcut for identify, I moved the download overlay into sidebar index.js with the other overlay functions. I also cleaned up the remaining overlay in file-operations.js, DeleteOverlay, by moving it into index.js. These might belong somewhere else but I thought I would follow the pattern we started. They are now taking advantage of Flux by using state to show and hide. It should be very easy to move these if we want a view file that holds all of the overlays, or something like that. In addition, I am passing the actions down as props into file-operations.js instead of requiring the actions in the component. I cleaned up any remaining actions in file-operations to follow this pattern as well, removing all actions from the component.
#### What are the relevant tickets?

Closes #157
Closes #123
